MalApi/MyAnimeListApi.cs

```diff
@@ -72,6 +78,40 @@ private HttpRequestMessage InitNewRequest(string uri, HttpMethod method)
             return request;
         }
 
+        private HttpRequestMessage InitNewRequestWithCredentials(string user, string password, string uri, HttpMethod method, UpdateObjectBase updateInfo = null)
+        {
+            HttpRequestMessage request = new HttpRequestMessage(method, uri);
+
+            // Requests with credentials require them to be encoded in base64
+            string credentials;
+            if (user != null && password != null)
+            {
+                var plainTextBytes = Encoding.UTF8.GetBytes(user + ":" + password);
+                credentials =  Convert.ToBase64String(plainTextBytes);
+            }
+            else
+            {
+                throw new Exception("Invalid input credentials format (both username and password have to be input OR just the base64 representation");
+            }
+
+            if (UserAgent != null)
+            {
+                request.Headers.TryAddWithoutValidation("User-Agent", UserAgent);
+                // Adding the authorization header with the credentials
+                request.Headers.TryAddWithoutValidation("Authorization", "Basic " + credentials);
```

Comment:
Why only set the Authorization header if the UserAgent property is set?

Reply:
Using header properties that the HttpRequestMessage API supports directly is preferable to using TryAddWithoutValidation. TryAddWithoutValidation is used for User-Agent because it complains if the user agent you set does not follow the same format browsers use, which is not the case when it is being (mis)appropriated as an API key. For authorization you should be able to use

```
request.Headers.Authorization = new AuthenticationHeaderValue("Basic", credentials);
```